### ISSUE TITLE
How to calculate the correct in ImageRecognition/main_Bayes.py

### ISSUE THREAD

**Issue by CassieMai** [kumar-shridhar/PyTorch-BayesianCNN#30]
@kumar-shridhar Hi, when testing the alexNet model, I have the error about the size miss matching between correct and preds. How to calculate the correct? In your code, correct += preds. Should it be correct += preds.size(0)? When I use correct += preds.size(0), the test accuracy is 100%. How to calculate the correct?

```
def test(epoch):
    global best_acc
    net.eval()
    test_loss = 0
    correct = 0
    total = 0
    conf=[]
    m = math.ceil(len(testset) / cf.batch_size)
    for batch_idx, (inputs_value, targets) in enumerate(testloader):
        x = inputs_value.view(-1, inputs, resize, resize).repeat(cf.num_samples, 1, 1, 1)
        y = targets.repeat(cf.num_samples)
        if use_cuda:
            x, y = x.cuda(), y.cuda()
        with torch.no_grad():
            x, y = Variable(x), Variable(y)
            outputs, kl = net.probforward(x)
     ...

        loss = vi(outputs,y,kl,beta)

        test_loss += loss.data.item()
        _, predicted = torch.max(outputs.data, 1)
        preds = F.softmax(outputs, dim=1)
        results = torch.topk(preds.cpu().data, k=1, dim=1)
        #print(results[0][0].item())
        conf.append(results[0][0].item())
        total += targets.size(0)
        correct += preds
        print('total: ', total)
        print('correct:', correct)
        predicted.eq(y.data).cpu().sum()
```

**Comment by kumar-shridhar** [comment]
Hi @CassieMai ,

This is not the best-written file and might be buggy. Please use the Jupyter notebook `ImageRecognition/BayesianCNN_detailed.ipynb` to see the latest working code. Hope it will solve the issue.

**Comment by CassieMai** [comment]
I see. Thank you.
